src/frontend/flight/handler.rs

```diff
@@ -73,14 +80,12 @@ impl SeafowlFlightHandler {
     // Get a specific stream from the map
     pub async fn fetch_stream(
         &self,
-        query_id: String,
+        query_id: &str,
     ) -> core::result::Result<SendableRecordBatchStream, Status> {
-        let (_, batch_stream_mutex) =
-            self.results
-                .remove(&query_id)
-                .ok_or(Status::not_found(format!(
-                    "No results found for query id {query_id}"
-                )))?;
+        let (_, batch_stream_mutex) = self.results.remove(query_id).ok_or_else(|| {
+            debug!("No results found for query id {query_id}");
```

Comment:
Should be an INFO/ERROR? If a client is requesting a query that it thought we were executing but we aren't, we definitely want to know